proto/connector/v1/connector.proto

```diff
@@ -8,220 +8,236 @@ import "google/protobuf/struct.proto";
-    // start writing records. If needed, the plugin should open connections in
-    // this function.
-    rpc Start(Destination.Start.Request) returns (Destination.Start.Response);
-    // Run will open a bidirectional stream between Conduit and the plugin.
-    // Conduit will be streaming records to the plugin that should be written
-    // to the 3rd party resource. The plugin is responsible for sending
-    // acknowledgments back to Conduit once a record has been processed. The
-    // acknowledgment should contain an error in case a record could not be
-    // successfully processed. The stream should still stay open in case Conduit
-    // is able to recover from the error and the pipeline keeps running.
-    rpc Run(stream Destination.Run.Request) returns (stream Destination.Run.Response);
-    // Stop signals to the plugin that no more records will be written to the
-    // stream. The plugin should flush any records that might be cached and not
-    // yet written to the 3rd party resource.
-    rpc Stop(Destination.Stop.Request) returns (Destination.Stop.Response);
-    // Teardown signals to the plugin that there will be no more calls to any
-    // other function. After Teardown returns, the plugin should be ready for a
-    // graceful shutdown.
-    rpc Teardown(Destination.Teardown.Request) returns (Destination.Teardown.Response);
+  // Configure is the first function to be called in a plugin. It provides the
+  // plugin with the configuration that needs to be validated and stored. In
+  // case the configuration is not valid it should return an error status.
+  rpc Configure(Destination.Configure.Request) returns (Destination.Configure.Response);
+  // Start is called after Configure to signal the plugin it can prepare to
+  // start writing records. If needed, the plugin should open connections in
+  // this function.
+  rpc Start(Destination.Start.Request) returns (Destination.Start.Response);
+  // Run will open a bidirectional stream between Conduit and the plugin.
+  // Conduit will be streaming records to the plugin that should be written
+  // to the 3rd party resource. The plugin is responsible for sending
+  // acknowledgments back to Conduit once a record has been processed. The
+  // acknowledgment should contain an error in case a record could not be
+  // successfully processed. The stream should still stay open in case Conduit
+  // is able to recover from the error and the pipeline keeps running.
+  rpc Run(stream Destination.Run.Request) returns (stream Destination.Run.Response);
+  // Stop signals to the plugin that the record with the specified position is
+  // the last one and no more records will be written to the stream after it.
+  // Once the plugin receives the last record it should flush any records that
+  // might be cached and not yet written to the 3rd party resource.
+  rpc Stop(Destination.Stop.Request) returns (Destination.Stop.Response);
```

Comment:
This comment changed, the stop request now contains the last record position, giving the plugin the ability to determine when it received all records and when to flush them.